nodl_python/nodl/_index.py

```diff
@@ -65,7 +65,26 @@ def get_node_by_executable(*, package_name: str, executable_name: str) -> Node:
     """
     nodes = _get_nodes_from_package(package_name=package_name)
     try:
-        result = next(node for node in nodes if node.executable == executable_name)
+        return next(node for node in nodes if node.executable == executable_name)
     except StopIteration:
         raise ExecutableNotFoundError(package_name=package_name, executable_name=executable_name)
-    return result
+
+
+def get_nodes_by_executables(
```

Comment:
If you're requesting multiple nodes, you'll want to use this rather than re-parse every time in the singular method. It could go in ros2nodl as a helper function but I can see others wanting to use it. I can remove it if need be.

Reply:
Let's wait and see if it's needed beyond the CLI. It's easy to make something public later, not so easy to make something non-public later.

Reply:
fixed in 1d45ece